File: leetcode/0477-total-hamming-distance/solution.py
from typing import List


class Solution:
    def totalHammingDistance(self, nums: List[int]) -> int:
        n = len(nums)
        total = 0
        for bit in range(30):
            ones = sum((num >> bit) & 1 for num in nums)
            total += ones * (n - ones)
        return total

File: leetcode/0477-total-hamming-distance/Solution.java
class Solution {
    public int totalHammingDistance(int[] nums) {
        int n = nums.length;
        long total = 0;
        for (int bit = 0; bit < 30; bit++) {
            int ones = 0;
            for (int num : nums) {
                ones += (num >> bit) & 1;
            }
            total += (long) ones * (n - ones);
        }
        return (int) total;
    }
}
